### DISCUSSION TITLE
Is it possible to attach client to specific workspace?

### DISCUSSION THREAD

**Discussion by doganalper** [VonHeikemen/lsp-zero.nvim#149] [Q&A]
I'm frontend developer who uses vue on multiple repos and on some I use Volar on some I use Vetur. Is it possible to attach volar on some workspaces and same for vetur?

**Comment by VonHeikemen** [comment]
The easiest solution I can think of is creating a file for volar in your root directory.

```lua
local lsp = require('lsp-zero')
lsp.preset('recommended')

local root_pattern = require('lspconfig').util.root_pattern

lsp.configure('volar', {
  root_dir = root_pattern('package.json', '.volar')
})

lsp.setup()
```

This way `vuels` will attach when the root directory has a `package.json` and `vue.config.js`. And volar will attach when it finds `package.json` and `.volar`.

Now, `.volar` is not a special file. The suggestion I make here is just to create an empty file and tell neovim to look for it.

**Reply by doganalper** [reply]
I managed to make this work by adding code below, vue2 projects I work with has `vetur.config.js` and vue3 projects have files about vite.
```lua
lsp.configure('vuels', {
	root_dir = root_pattern('vetur*')
})

lsp.configure('volar', {
	root_dir = root_pattern('vite*')
})
```
I might be wrong but `root_pattern` might work like `or`, because when I have `package.json` as parameter, on vue2 project it still attaches `volar`. Thats why I didn't include `package.json` to final code.

**Reply by doganalper** [reply]
Eventually my plan is reading package.json, parsing vue version and attach depending on it but I need to learn more about lua and lsp before that.
